src/hooks/useCartItems.ts

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect } from 'react';
+import { addCartItem, deleteCartItem, fetchCartItem } from '../api';
+import { CartItemType } from '../types';
+import { useToast } from './useToast';
+import { ERROR } from '../constant/message';
+
+export function useCartItems() {
+  const [cartItems, setCartItems] = useState<CartItemType[]>([]);
+  const [idMap, setIdMap] = useState<Record<number, number>>({});
```

Comment:
```suggestion
export function useCartItems() {
  const [cartItems, setCartItems] = useState<CartItemType[]>([]);
  const [idMap, setIdMap] = useState<Record<ProductType['id'], CartItemType['id']>>({});
```

어떤 id를 어떤 id에 맵핑하는 건지 명시해주면 더 이해하기 좋을 듯 합니다.